pkg/plumbing/watch/watch.go

```diff
@@ -218,65 +164,71 @@ func canonicalizePath(pwd, path string) string {
 //    - warpforge-error-serialization -- when the module or plot cannot be parsed
 //    - warpforge-error-unknown -- when changing directories fails
 //    - warpforge-error-unknown -- when context ends for reasons other than being canceled
-//    - warpforge-error-initialization -- unable to get working directory or executable path
+//    - warpforge-error-searching-filesystem --
+//    - warpforge-error-module-invalid --
 func (c *Config) Run(ctx context.Context) error {
 	log := logging.Ctx(ctx)
+	searchPath := canonicalizePath(c.WorkingDirectory, c.Path)
 
-	wd, xerr := os.Getwd()
-	if xerr != nil {
-		return serum.Error(wfapi.ECodeInitialization, serum.WithCause(xerr), serum.WithMessageLiteral("unable to get working directory"))
+	log.Debug("", "search path: %s", searchPath)
+	ws, _, err := workspace.FindWorkspace(c.Fsys, "", searchPath)
+	if err != nil {
+		return err
 	}
-
-	modulePath := filepath.Join(c.Path, dab.MagicFilename_Module)
-	modulePathAbs := canonicalizePath(wd, modulePath)
-
-	// TODO: currently we read the module/plot from the provided path.
-	// instead, we should read it from the git cache dir
-	plot, err := dab.PlotFromFile(os.DirFS(c.Path), dab.MagicFilename_Plot)
+	_, wsPath := ws.Path()
```

Comment:
`FindWorkspace` returns nil if no workspace is found. This will cause a nil pointer deref.

Reply:
I had an old `.warpforge` dir in my home directory hiding this. :scream: